### PR TITLE
DEP 11 -- Add accessibility team draft DEP

### PR DESCRIPTION
**Pull request by knyghty**
I think this is about as far as I can get with this myself. The area I need the most help with is related to governance and processes, and I have little knowledge of how other, similar teams work.

Thanks to @thibaudcolas for the early feedback.

### CODE REVIEW

**Comment by thibaudcolas** [comment]
Link back to corresponding django-developers discussion for future reference: [`django-developers` Admin accessibility](https://groups.google.com/g/django-developers/c/FsBrNGTxvCA).

File: draft/accessibility-team.rst [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+The Accessibility Team (informally, a11y team) shall be formed.
+
+Membership
+----------
+
+The team shall not have a fixed size, but instead will grow and shrink
+organically as members choose to leave, and when new members are deemed to be
+required by the rest of the team. Membership of the team works similarly to the
+`Code of Conduct Committe <https://github.com/django/code-of-conduct/blob/master/membership.md>`_.
```
Comment (tomdyson) [inline draft/accessibility-team.rst:37]: Committe -> Committee

File: draft/accessibility-team.rst [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+- Have disabilities that make using the web and/or web development more
+  difficult.
+- Have relevant, positive contributions or expertise in the field.
+- Have a record of contributing to Django.
+
+Members shall remain in the team for a fixed-term of 9 months, after which
+they must opt-in to remain on the team. Membership will also be terminated by:
+
+- Becoming disqualified due to actions taken by the Code of Conduct committe
```
Comment (tomdyson) [inline draft/accessibility-team.rst:50]: committe -> committee

File: draft/accessibility-team.rst [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+Responsibilities
+----------------
+
+The remit of the Accessibility Team shall include all user-facing components
+of projects maintained by the Django Software Foundation. This includes but is
+not limited to:
+
+- User-facing parts of Django, such as default HTML output in forms and such,
+  and the django admin and its default theme, HTML, and UI components.
```
Comment (tomdyson) [inline draft/accessibility-team.rst:65]: django -> Django

File: draft/accessibility-team.rst [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+Improving accessibility also has a tendency to improve the experience of other
+users. This is called the
+`curb cut effect <https://alexwlchan.net/2019/01/monki-gras-the-curb-cut-effect/>`_
+While this is mostly noted in the real world, the web also experiences this
+effect. For example, keyboard shortcuts are used to allow keyboard-only users
+navigate a site more efficiently, but this positive benefit affects everyone,
+not just the physically disabled.
+
+The django admin currently exhibits many accessibility issues, such as forms
```
Comment (tomdyson) [inline draft/accessibility-team.rst:119]: django -> Django

File: draft/accessibility-team.rst [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+fields missing labels, low color contrast, a lacklustre keyboard-only
+experience among other issues. Other parts of the project have problems such
```
Comment (tomdyson) [inline draft/accessibility-team.rst:121]: contrast, a lacklustre keyboard-only experience among other issues -> contrast and a lacklustre keyboard-only
experience

**Review by tomdyson — COMMENTED** [review]
Thanks for this @knyghty!

**Comment by knyghty** [comment]
Thanks @tomdyson - fixed those.

I've also removed the reference to Django People as it's been sunset now.

**Review by carltongibson — COMMENTED** [review]
Hi @knyghty @thibaudcolas. 
So... 🙂

Giving this a read over again now, thinking particularly about how Mariusz and I will deal with this on a day-to-day basis. I wonder...

Is it feasible to say in a bit more detail about HOW we maintain accessibility? i.e. the Rationale sections reads little light. 

**Possibly** the answer there is No, we just need to push on (part of it is in the working out) but "It's feasible to do automated checks to this extent", "Areas in which a11y team should be asked for review include...", and so on might give more structure/guidance. 

Then my only other concern is _staffing_ — we need to advertise and promote the team to make sure we can get enough people on-board, on a rolling basis. (Possibly that's out of scope for this DEP — ties into discussions about recognizing contributors more generally.) 

Any other issues?

What do you two feel is left to do before we push forwards and ask the TB to consider?

**Comment by knyghty** [comment]
@carltongibson --

My feeling is that we should leave the DEP fairly non-specific and allow the a11y team to define their own role (with the TB being able to nullify whatever parts of that they wish), which may need to change over time as things change. I don't want to codify it too much in the DEP because while the DEP can change, there is quite a bit of friction with that I think, it's quite a slow process.

The staffing concern is one I share, in the DEP it states that recruitment will work more or less as the CoC team does, advertised with blog posts and so on -- hopefully that is enough, but time will tell.

The only things remaining on my mind are:

1. If not in this DEP, then where are those guidelines set out? e.g. when should you request a review from a11y team. My current feeling is the contributing guide review checklist. I'm not (yet) sure if there are other similar questions.

2. Do the various processes laid out here seem reasonable (regarding TB input, the relative independence of the team, etc.)?

3. How does the a11y team communicate with everyone else. e.g. we probably need some help from the ops team with Jenkins integration, if we go a route involving CI, or how does the TB stay up to date so they can keep the various factors balanced -- probably a mailing list, but maybe this whole question is just an implementation detail.

**Comment by thibaudcolas** [comment]
Hey both, I don’t think I’ll have much of a useful take on this having been involved with Django development so little. But I’ll try.

For the “How we maintain accessibility” – it’ll be a combination of manual and automated checks, that either are available for contributors to self-serve, or for reviewers, or auto-run in CI, or separate monitoring. The DEP could have examples of these, but I think it’s good to keep it open. There are two things that I’d say are almost givens and could be included if desired:

- There will be new guidelines for manual testing as part of contributing resources. How extensive these are, and how much of expertise these will require – to be decided.
- It will be highly desirable to have a baseline of automated checks, and make sure that no contribution fails those checks.

For staffing – if we can get an initial group of 3+ people, then I’d hope that group’s work will raise awareness of accessibility considerations with the Django community enough to find its future contributors. Even if the team was to disband relatively quickly – I’d hope our initial work can be enough so the accessibility support targets, review guidelines, and automated tests could be sustained by the rest of the project.

For what might be remaining to decide – I don’t think any are blockers but what’s on my mind is:

- Which standard(s) we’d want Django to aim for – whether it’s WCAG2.1 or ATAG 2.0 (which includes WCAG).
- For the start of the team, how we decide what to do / what’s the most impactful – there’ll be a lot to fix between parts of Django that affect end users, the admin, the docs, and websites maintained by the project.

**Comment by carltongibson** [comment]
Hey both. Thanks for you comments. 

* I think deferring some points is fine. 
* Let's ask for input on some of the minor points: It shouldn't be the case that not knowing the smaller details blocks the big idea.
* "what’s the most impactful": Admin, then docs site, then other stuff. (I'd say.)

@knyghty DEP1 says I have to give it a number, so DEP 11 shall we say. Can you rename files, fill in the blanks, and such? I'll then give it a last read and then push it to django-developers for advancement. 

Thanks!

**Comment by knyghty** [comment]
@carltongibson Thanks, I've renamed the file and filled in the numbers. Let me know if there's anything else.

**Comment by carltongibson** [comment]
I don't think so. Let's leave it for the weekend to see if there's further input. If not we'll push it onwards. 

Thanks for taking the initiative here both.

File: draft/0011-accessibility-team.rst [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+.. contents:: Table of Contents
+  :depth: 3
+  :local:
+
+Abstract
+========
+
+This DEP proposes the formation of an Accessibility Team to encourage projects
+maintained by the Django Software Foundation to be accessible to as diverse a
+group as possible, particularly people with disabilities that make using the
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:23]:
```suggestion
maintained by the Django Software Foundation to be accessible to as many
people as possible, particularly those with disabilities that make using the
```

File: draft/0011-accessibility-team.rst [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+of volunteers, an advertisement will be published on the Django website.
+Priority will be given to volunteers who, in no particular order:
+
+- Have disabilities that make using the web and/or web development more
+  difficult.
+- Have relevant, positive contributions or expertise in the field.
+- Have a record of contributing to Django.
+
+Members shall remain in the team for a fixed-term of 9 months, after which
+they must opt-in to remain on the team. Membership will also be terminated by:
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:48]: Will opting-in renew for another 9 months? Will team members need to continue opting-in each 9 months in perpetuity?

File: draft/0011-accessibility-team.rst [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+The remit of the Accessibility Team shall include all user-facing components
+of projects maintained by the Django Software Foundation. This includes but is
+not limited to:
+
+- User-facing parts of Django, such as default HTML output in forms and such,
+  and the Django admin and its default theme, HTML, and UI components.
+
+- All websites and other software projects maintained by the DSF, such as the
+  Django website, including documentation, Django Snippets, the issue tracker,
+  etc.
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:69]: I think the team will cover *all* documentation right, no just that on the Django website - e.g. Channels on Read the Docs ?

```suggestion
  Django website, projects' documentation, Django Snippets, the issue tracker, etc.
```

File: draft/0011-accessibility-team.rst [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+
+The responsibilities of the Accessibility Team are expected to change over
+time, and are to be decided by consensus of the team, with input from the
+Technical Board as required. To begin, several areas have been identified:
+
+- Deciding on any relevant accessibility guidelines to follow, such as WCAG,
+  and at which conformance level.
+
+- Implementing automated testing to catch issues, working with the ops
+  team as needed to integrate this into the CI process.
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:79]: nitpick - though I guess most work will be on Django core, multiple projects = multiple CI proceses
```suggestion
  team as needed to integrate this into CI processes.
```

File: draft/0011-accessibility-team.rst [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+- Coordinating the fixing of accessibility issues and the improvement of the
+  accessibility in general in Django and associated projects.
+
+- Writing and maintaining documentation relating to accessibility, such as
+  a statement of commitment to accessibility issues, and contribution
+  guidelines.
+
+- Reviewing accessibility fixes, improvements and other tickets that may affect
+  the accessibility of the project.
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:91]:
```suggestion
  accessibility of any relevant project.
```

File: draft/0011-accessibility-team.rst [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+=========
+
+An alternative is to go on as usual, leaving accessibility in the hands of
+individual contributors when writing and reviewing code. However, given the
+accessibility issues that exist in the Django admin and on the
+`Django website <https://www.djangoproject.com/>`__, this doesn't seem to be
+enough.
+
+Another option is to implement some basic standards, such as conforming to WCAG
+and setting up a CI to test. This approach is better than nothing but it
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:137]:
```suggestion
and setting up a CI tests. This approach is better than nothing but it
```

File: draft/0011-accessibility-team.rst [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+Specification
+=============
+
+The Accessibility Team (informally, a11y team) shall be formed.
+
+Membership
+----------
+
+The team shall not have a fixed size, but instead will grow and shrink
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:34]: Should we spell out in this section that Tom and Thibaud are the first members? Are there any other first members?

File: draft/0011-accessibility-team.rst [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+group as possible, particularly people with disabilities that make using the
+web more difficult.
+
+Specification
+=============
+
+The Accessibility Team (informally, a11y team) shall be formed.
+
+Membership
+----------
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:32]: Unsure if this needs spelling out, but presumably the team will be listed on https://www.djangoproject.com/foundation/teams/  .

File: draft/0011-accessibility-team.rst [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+  a statement of commitment to accessibility issues, and contribution
+  guidelines.
+
+- Reviewing accessibility fixes, improvements and other tickets that may affect
+  the accessibility of the project.
+
+To aid in reviewing, the Accessibility Team members shall be added to a team
+in the GitHub organization with read access to relevant repositories, so that
+they may be requested to review pull requests, at the discretion of the author,
+reviewer, or other party.
```
Comment (adamchainz) [inline draft/0011-accessibility-team.rst:96]: Can we specify the name of that github team here as (`accessibility`?), so anyone reading the DEP can see the name to request on their PR's without having to hunt elsewhere?

**Review by adamchainz — COMMENTED** [review]
Nice work. I'm really excited for this. I mostly have some nitpicky-level comments/suggestions, and a few questions around extra information on membership. Otherwise, loving this and thank you for putting in the time to do a DEP and volunteering to improve a11y for Django!

File: draft/0011-accessibility-team.rst [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+
+Members shall remain in the team for a fixed-term of 9 months, after which
+they must opt-in to remain on the team. Membership will also be terminated by:
+
+- Becoming disqualified due to actions taken by the Code of Conduct committee
+  of the Django Software Foundation.
+
+- A vote of the Technical Board, or full consensus of the rest of the
+  Accessibility Team, if the team is considered too large, the person is not
+  making positive contributions, or any other sound reason.
```
Comment (orf) [inline draft/0011-accessibility-team.rst:55]: Is it worth explicitly stating that they can also terminate their own membership?

File: draft/0011-accessibility-team.rst [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+- Coordinating regular manual accessibility audits on all relevant projects.
+
+- Coordinating the fixing of accessibility issues and the improvement of the
+  accessibility in general in Django and associated projects.
+
+- Writing and maintaining documentation relating to accessibility, such as
+  a statement of commitment to accessibility issues, and contribution
+  guidelines.
+
+- Reviewing accessibility fixes, improvements and other tickets that may affect
```
Comment (orf) [inline draft/0011-accessibility-team.rst:90]: I think adding an explicit goal of identifying areas where the framework could provide API improvements/new features that help users of Django also create accessible sites would be good?

This is kind of covered by this statement, but rather implicitly

File: draft/0011-accessibility-team.rst (hunk 13 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst:32]: I'm also unsure. It seems a bit unnecessary and I also couldn't think of a good way to word it into the DEP.

File: draft/0011-accessibility-team.rst (hunk 12 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst:34]: I don't feel any particular need. I don't see a reason why initial members should be prioritised over anyone else, and we're in the implementation team anyway.

File: draft/0011-accessibility-team.rst (hunk 7 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst]: Yes. I've clarified the wording a bit here.

File: draft/0011-accessibility-team.rst (hunk 15 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst:57]: I think it is. I've added it.

File: draft/0011-accessibility-team.rst (hunk 16 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst:91]: Yes, that's a good point, and I've added it.

File: draft/0011-accessibility-team.rst (hunk 14 above)
Comment (knyghty) [reply draft/0011-accessibility-team.rst:97]: Done.

File: draft/0011-accessibility-team.rst (hunk 13 above)
Comment (adamchainz) [reply draft/0011-accessibility-team.rst:32]: Fair.

**Review by adamchainz — APPROVED** [review]

**Review by carltongibson — APPROVED** [review]
Hi @knyghty @thibaudcolas — sorry for the slow movement forward here — such is how it goes. 

This is great. Thanks so much for your work here, and for taking the initiative to push things forward. 

Based on the mailing list discussion, I've updated, squashed, and moved to `accepted`. I'll now merge and create the GitHub team so we can easily request your input on PRs. 

I'll also look into adding an entry to the [Teams page of the site](https://www.djangoproject.com/foundation/teams/). 

I think a PR to the Contributing Guide in the initial [several ways you can contribute](https://docs.djangoproject.com/en/3.1/internals/contributing/) list saying something like...

```
* Join the accessibility team — do you have disabilities that make using the web
  and/or web development more difficult, or relevant expertise in the field? Say  
  [hello on the forum](link to forum) and help us to make sure that app built with 
  Django can be accessible by default. 
```

... or similar might be a good addition. 🤔

Thanks again! 🎩

**Comment by thibaudcolas** [comment]
This makes my day, my week, and probably the whole month as well! Amazing! Thank you again @knyghty for getting the proverbial ball rolling.

I have no idea what we’ll be doing next to kick things off, but looking forward to it all regardless! 😄

**Comment by knyghty** [comment]
Great to see this 🚀 

I suppose a good first step to implementing this DEP is to figure out some venue for discussion. I'm open to suggestions. The forum might be a natural place, though I don't currently read it very often, or a mailing list might also be good. Either will be very low volume I think, but we need some place to get together and figure out ideas for CI, general improvements, and what have you.

**Comment by thibaudcolas** [comment]
How do other teams coordinate? I imagine that even though it’s just the two of us currently, we’d rather follow a set pattern to make it as easy as possible for others to contribute to the conversations regardless of their team membership.

Personally I’m happy with whichever channel works for others really.

**Comment by carltongibson** [comment]
> How do other teams coordinate? 

I think it's pretty informal. — We have the GitHub team, which had discussions. I'd suggest using that, since you get a notification and such. Other than that, IRC, Forum, ... — and then communicating on issues. 

(Good bump too: I need to resolve adding the team to the website. Note to self. )

**Comment by thibaudcolas** [comment]
Ping @knyghty for thoughts on this :) Personally I think we should favour something that makes it easy for others to contribute to discussions, or at least keep tabs of what the team does. Particularly people that aren’t already core contributors to Django, so we can get more accessibility-specific feedback when needed.

Looking at GitHub teams discussions ([test discussion](https://github.com/orgs/django/teams/accessibility/discussions/1)), it doesn’t seem like there’s a way to make those public, so I’d be more inclined for a forum or mailing list.

For what it’s worth I’ve also registered to the #django IRC channel. First time using IRC ever :)

**Comment by knyghty** [comment]
Hmm I'd tend to agree that something public would be useful. #django more concerns usage, there is also #django-dev but it's very low volume and I think it's debatable how "public" it really is since there are no public logs anymore.

I think something on the forum is probably good. Although I don't really use it at the moment, it's probably easier to maintain than another mailing list.

**Comment by carltongibson** [comment]
> I think something on the forum is probably good

Yep :+1: — I'd say to open a topic in the Contributing channel to discuss any issues.

**Comment by thibaudcolas** [comment]
For future ref – as of #82, DEP 11 is now "final". I wanted to do this so we have a stable link to the "latest" version which will never change – which is <https://github.com/django/deps/blob/main/final/0011-accessibility-team.rst>.

I’m sure everyone here watches other channels for contributors but just in case – the accessibility team is currently making good progress on _guidelines for Django contributors to follow_. See:

- [Accessibility guidelines for all contributors #17338](https://github.com/django/django/pull/17338)
- [django/django#17339](https://github.com/django/django/pull/17339)
- [django/django#17340](https://github.com/django/django/pull/17340)

And a call for feedback on said guidelines on the forum: [Feedback needed: accessibility guidelines for contributors](https://forum.djangoproject.com/t/feedback-needed-accessibility-guidelines-for-contributors/25437)

**Pull request merged**